### PR TITLE
T1064 Scripting

### PR DESCRIPTION
**Pull request by VakarisZ**
# Feature
> Implemented T1064 Scripting attack technique
image

### CODE REVIEW

File: monkey/infection_monkey/post_breach/pba.py [hunk 1]
```diff
@@ -46,6 +47,8 @@ def run(self):
         """
         exec_funct = self._execute_default
         result = exec_funct()
+        if result[1] and isinstance(self.command, list) and len(self.command) > 1:
+            T1064Telem(ScanStatus.USED, "Scripts used to execute %s post breach action." % self.name).send()
```
[the remark this answers is not in the document]
Comment (VakarisZ) [reply monkey/infection_monkey/post_breach/pba.py]: I refactored. See and tell if got more clear about what it does.

File: monkey/infection_monkey/post_breach/pba.py [hunk 2]
```diff
@@ -46,8 +47,21 @@ def run(self):
...
         PostBreachTelem(self, result).send()
 
+    def scripts_were_used(self, pba_execution_result):
+        """
+        Determines if scripts were used to execute PBA
+        :param pba_execution_result: result of execution function. e.g. self._execute_default
+        :return: True if scripts were used, False otherwise
+        """
+        pba_execution_succeeded = pba_execution_result[1]
+        if pba_execution_succeeded and isinstance(self.command, list) and len(self.command) > 1:
```
Comment (itaymmguardicore) [inline monkey/infection_monkey/post_breach/pba.py]: The following pattern:
```
if cond:
  return True
return False
```
should always be replaced with:
`return cond`

File: monkey/infection_monkey/post_breach/pba.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply monkey/infection_monkey/post_breach/pba.py]: This fixes the second part of the comment (which is the less significant one).
If your claim is that by definition having multiple commands here is a script by itself, I can accept that, but if that's your way of determining this was called from BackdoorUser (which actually uses scripting), then it's hacky and will probably not be correct in the future.
If it's the latter case, fix it, if it's the first then there are 2 problems:
1. We need to change the reported text
2. the function `scripts_were_used` returns True iff scripts were used and were run successfully. The second condition shouldn't be coupled with the first

File: monkey/infection_monkey/post_breach/pba.py (hunk 2 above)
Comment (VakarisZ) [reply monkey/infection_monkey/post_breach/pba.py]: God damn it. How do I keep forgetting that....

**Pull request merged**